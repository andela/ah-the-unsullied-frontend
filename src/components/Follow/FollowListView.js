import React, { Component } from 'react';
import { connect } from 'react-redux';
import { Card, Col, CardTitle, Row } from 'react-materialize';
import { Link } from 'react-router-dom';
import PropTypes from 'prop-types';
import {followingList} from '../../actions/profile/followActions';
import '../../assets/css/articles.scss';
import '../../assets/styles/HomePage.scss';
import Nav from '../common/nav/index';

export class FollowingListView extends Component {
  componentDidMount() {
    const username = this.props.match.params.username;
    this.props.followingList(username)
  }

  render() {
    const list = this.props.Following.following;
    const Fusername = this.props.match.params.username;
    const user_details = list.map(item => {
      const username = item.username
      return (
        <div>
          <Row>
            <Col m={7} s={12}>
              <Card
                horizontal
                header={<CardTitle image={item.image} />}
              >
                <Link
                  to={{
                    pathname: `/profile/${username}`,
                    username: { username }
                  }}
                >
                  <b>{item.username}</b>
                </Link>
                <div className="bio">
                  <p>{item.bio}</p>
                </div>
                <div className="follow" />
              </Card>
            </Col>
          </Row>
        </div>
      );
    });
    return (
      <div>
        <Nav />
        <div className="container">
          <h3> {Fusername} follows </h3>
        </div>

        {user_details}
      </div>
    );
  }
}

FollowingListView.propTypes = {
  Following: PropTypes.object.isRequired,
  followingList: PropTypes.func.isRequired,
  username: PropTypes.string.isRequired, 
  match: PropTypes.object.isRequired
};


export function mapStateToProps(state) {
  return {
    Following: state.following,
    auth: state.auth
  };
}

export default connect(
  mapStateToProps,
 {followingList}
)(FollowingListView);
