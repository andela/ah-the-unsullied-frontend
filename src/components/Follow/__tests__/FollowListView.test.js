import React from 'react';
import Enzyme, { shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16/build';
import jest from 'jest-mock';
import { FollowingListView, mapStateToProps, mapDispatchToProps } from '../FollowListView';

Enzyme.configure({ adapter: new Adapter() });
describe('followwing ListView', () => {
  let props;
  let wrapper;
  let wrapperInstance;
  let state;
  let dispatchProps;
  let dispatch;
  beforeEach(() => {
    props = {
      bio: 'i have abio',
      username: 'nesh',
      image: 'hshhshs',

      auth: {
        isAuthenticated: true,
        user: {
          username: 'nesh'
        }
      },
      Following: {
        following: []
      },

      match: {
        params: {
          username: 'nesh'
        }
      },

      list: [
        {
          bio: 'i have abio',
          username: 'nesh',
          image: 'hshhshs'
        },
        {
          bio: 'i have abio',
          username: 'nesh',
          image: 'hshhshs'
        }
      ],

      followingList: jest.fn(() => {
        Promise.resolve();
      })
    };

    state = {
      auth: {},
      Following: {
        following: [{},{}]
      }
    };

    wrapper = shallow(<FollowingListView {...props} />);
    wrapperInstance = wrapper.instance();
  });

  it('should render the follow button', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('mapStateToProps', () => {
    const propsState = mapStateToProps(state);
    expect(propsState.auth).toEqual(state.auth);
  });
});
