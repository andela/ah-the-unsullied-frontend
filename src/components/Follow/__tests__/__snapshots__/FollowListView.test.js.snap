// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`followwing ListView should render the follow button 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <FollowingListView
    Following={
      Object {
        "following": Array [],
      }
    }
    auth={
      Object {
        "isAuthenticated": true,
        "user": Object {
          "username": "nesh",
        },
      }
    }
    bio="i have abio"
    followingList={
      [MockFunction] {
        "calls": Array [
          Array [
            "nesh",
          ],
        ],
        "results": Array [
          Object {
            "type": "return",
            "value": undefined,
          },
        ],
      }
    }
    image="hshhshs"
    list={
      Array [
        Object {
          "bio": "i have abio",
          "image": "hshhshs",
          "username": "nesh",
        },
        Object {
          "bio": "i have abio",
          "image": "hshhshs",
          "username": "nesh",
        },
      ]
    }
    match={
      Object {
        "params": Object {
          "username": "nesh",
        },
      }
    }
    username="nesh"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <withRouter(Connect(Nav)) />,
        <div
          className="container"
        >
          <h3>
             
            nesh
             follows 
          </h3>
        </div>,
        Array [],
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <h3>
             
            nesh
             follows 
          </h3>,
          "className": "container",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              " ",
              "nesh",
              " follows ",
            ],
          },
          "ref": null,
          "rendered": Array [
            " ",
            "nesh",
            " follows ",
          ],
          "type": "h3",
        },
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <withRouter(Connect(Nav)) />,
          <div
            className="container"
          >
            <h3>
               
              nesh
               follows 
            </h3>
          </div>,
          Array [],
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <h3>
               
              nesh
               follows 
            </h3>,
            "className": "container",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                " ",
                "nesh",
                " follows ",
              ],
            },
            "ref": null,
            "rendered": Array [
              " ",
              "nesh",
              " follows ",
            ],
            "type": "h3",
          },
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
  Symbol(enzyme.__childContext__): null,
}
`;
